samples/Arducam/app/application.cpp

```diff
@@ -280,8 +285,8 @@ void init()
 	WifiStation.config(WIFI_SSID, WIFI_PWD);
 	WifiAccessPoint.enable(false);
 
-	// Run our method when station was connected to AP
-	WifiStation.waitConnection(connectOk);
+	WifiEvents.onStationConnect(&connectOk);
```

Comment:
Is there a reason to the callback params by reference? If I am not mistaken delegates should take care to handle those cases.